corehq/apps/users/models_role.py

```diff
@@ -196,7 +196,12 @@ def _clear_query_cache():
         _clear_query_cache()
 
     def get_permission_infos(self):
-        return [rp.as_permission_info() for rp in self.rolepermission_set.all()]
+        try:
+            return [rp.as_permission_info() for rp in self.rolepermission_set.all()]
+        except ValueError:
```

Comment:
`ValueError` is a generic exception that could be raised for other reasons than accessing a reverse foreign key relationship. Is there a more specific exception that can be caught? Alternatively, would it work to wrap only `self.rolepermission_set.all()` or a subset of it in the try/except?

It is not clear to me what is actually raising the error. Is it raised by `self.rolepermission_set`, `self.rolepermission_set.all()` or on iterating over the return value of the latter? Whatever it is, if we must catch `ValueError` we should limit the try/except to the narrowest scope possible to avoid catching an error raised somewhere else.

Reply:
Yeah good feedback. I'll wrap only what is needed, but unfortunately there isn't a more specific error to catch.

Reply:
Sorry I force pushed because of a sloppy commit, but only changed the most recent commit that you hadn't yet reviewed: 4b84b5337ececa5f7fd83fb9f7860ce7961dbe28 is the best I could do